src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -251,6 +209,33 @@ StreamReader::StreamReader(
                   std::move(stream_reader_data_context)
           )},
           m_ts_pattern{m_stream_reader_data_context->get_deserializer().get_timestamp_pattern()} {}
+
+auto StreamReader::create_deserializer_and_data_context(
+        std::unique_ptr<clp::streaming_compression::zstd::Decompressor>&& zstd_decompressor,
```

Comment:
Do you need to use `&&`? Maybe I am wrong, but it seems a `&&` in the function parameter means a function can take the reference of a rvalue. But since the input is a `std::move` of lvalue, we can simply take in the argument by value?

Reply:
I think with the `&&` we can enforce usage of `std::move` when invoking this method; otherwise, it's not strictly neccesary.

Reply:
I see, it's fair.